src/middleware/github-server-app-middleware.ts

```diff
@@ -1,31 +1,52 @@
 import { NextFunction, Request, Response } from "express";
 import { GitHubServerApp } from "models/github-server-app";
 import { Installation } from "models/installation";
+import { validationResult } from "express-validator";
+import { envVars } from "config/env";
 
 export const GithubServerAppMiddleware = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+
 	const { jiraHost } = res.locals;
-	const { id } = req.params;
+	const { uuid } = req.params;
+
+	req.log = req.log.child({ uuid, jiraHost });
 
-	if (id) {
-		req.log.debug(`Retrieving GitHub app with id ${id}`);
-		const gitHubServerApp = await GitHubServerApp.getForGitHubServerAppId(Number(id));
+	if (uuid && validationResult(req).isEmpty()) {
+		// uuid param exist, must be github server app
+		req.log.debug(`Retrieving GitHub app with uuid ${uuid}`);
+		const gitHubServerApp = await GitHubServerApp.findForUuid(uuid);
 
 		if (!gitHubServerApp) {
-			req.log.error({ id, jiraHost }, "No GitHub app found for provided id.");
+			req.log.error("No GitHub app found for provided uuid.");
 			throw new Error("No GitHub app found for provided id.");
 		}
-
 		const installation = await Installation.findByPk(gitHubServerApp.installationId);
-
 		if (installation?.jiraHost !== jiraHost) {
-			req.log.error({ id, jiraHost }, "Jira hosts do not match");
+			req.log.error({ uuid, jiraHost }, "Jira hosts do not match");
 			throw new Error("Jira hosts do not match.");
 		}
-
 		req.log.info("Found GitHub server app for installation");
-		res.locals.gitHubAppId = id;
-		return next();
+		res.locals.gitHubAppId = gitHubServerApp.appId;
```

Comment:
This is for compatibility for GHE pages. Even though they are not available for public yet, I don't want to break those pages or have a huge PR to refactor those.